internal/repo/answer/AddAnswer_test.go

```diff
@@ -0,0 +1,258 @@
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"xorm.io/xorm"
+)
+
+type mockUniqueIDRepo struct {
+	mock.Mock
+}
+
+func (m *mockUniqueIDRepo) GenUniqueIDStr(ctx context.Context, key string) (uniqueID string, err error) {
+	args := m.Called(ctx, key)
+	return args.String(0), args.Error(1)
+}
+
+type mockDBEngine struct {
+	mock.Mock
+}
+
+func (m *mockDBEngine) Context(ctx context.Context) *xorm.Session {
+	args := m.Called(ctx)
+	return args.Get(0).(*xorm.Session)
+}
+
+func (m *mockDBEngine) Insert(answer *entity.Answer) (int64, error) {
+	args := m.Called(answer)
+	return args.Get(0).(int64), args.Error(1)
+}
+
+func TestAddAnswer(t *testing.T) {
+	tests := []struct {
+		name           string
+		answer         *entity.Answer
+		enableShortID  bool
+		genUniqueIDErr error
+		insertErr      error
+		wantErr        error
+	}{
+		{
+			name: "Successfully add a new answer",
+			answer: &entity.Answer{
+				QuestionID:   "1234567890",
+				UserID:       "9876543210",
+				OriginalText: "Sample answer text",
+				ParsedText:   "Sample parsed answer text",
+			},
+			enableShortID:  false,
+			genUniqueIDErr: nil,
+			insertErr:      nil,
+			wantErr:        nil,
+		},
+		{
+			name: "Error generating unique ID",
+			answer: &entity.Answer{
+				QuestionID:   "1234567890",
+				UserID:       "9876543210",
+				OriginalText: "Sample answer text",
+				ParsedText:   "Sample parsed answer text",
+			},
+			enableShortID:  false,
+			genUniqueIDErr: errors.New("failed to generate unique ID"),
+			insertErr:      nil,
+			wantErr:        errors.New("failed to generate unique ID"),
+		},
+		{
+			name: "Error inserting answer into the database",
+			answer: &entity.Answer{
+				QuestionID:   "1234567890",
+				UserID:       "9876543210",
+				OriginalText: "Sample answer text",
+				ParsedText:   "Sample parsed answer text",
+			},
+			enableShortID:  false,
+			genUniqueIDErr: nil,
+			insertErr:      errors.New("failed to insert answer"),
+			wantErr:        errors.New("failed to insert answer"),
+		},
+		{
+			name: "Enable short ID functionality",
+			answer: &entity.Answer{
+				QuestionID:   "1234567890",
+				UserID:       "9876543210",
+				OriginalText: "Sample answer text",
+				ParsedText:   "Sample parsed answer text",
+			},
+			enableShortID:  true,
+			genUniqueIDErr: nil,
+			insertErr:      nil,
+			wantErr:        nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			if tt.enableShortID {
+				ctx = context.WithValue(ctx, constant.ShortIDFlag, true)
+			}
+
+			mockUniqueIDRepo := new(mockUniqueIDRepo)
+			mockUniqueIDRepo.On("GenUniqueIDStr", ctx, tt.answer.TableName()).Return("12345", tt.genUniqueIDErr)
+
+			mockDB := new(mockDBEngine)
+			mockDB.On("Context", ctx).Return(&xorm.Session{})
+			mockDB.On("Insert", mock.Anything).Return(int64(1), tt.insertErr)
+
+			ar := &answerRepo{
+				data:         &data.Data{DB: mockDB},
```

Comment:
cannot use mockDB (variable of type *mockDBEngine) as *xorm.Engine value in struct literal

only one compilation error.